### PR TITLE
feat: enable multi page star fetching for private vercel instances

### PR DESCRIPTION
**Pull request by rickstaa**
This PR enables multi-page stars' support from fetching on private Vercel instances. This feature can be disabled on the public Vercel instance but can be enabled by adding the `FETCH_MULTI_PAGE_STARS=true` as an env variable.

### CODE REVIEW

**Comment by rickstaa** [comment]
As pointed out by @anuraghazra. Before we merge this we have to make sure that the `maxDuration` limit of the private Vercel instance on the free plan is not being hit (see https://vercel.com/blog/customizing-serverless-functions).

**Comment by rickstaa** [comment]
@anuraghazra We can switch to `FETCH_MULTI_PAGE_STARS` and document that users can achieve more accurate results when they add this Vercel env parameter. Although less of a problem than in #2111 and #1691, we can also mention that for some users, this might lead to a Vercel timeout because of the [maxDuration]( https://vercel.com/blog/customizing-serverless-functions). Whatever you think is better.

**Comment by rickstaa** [comment]
The old code implemented in #2100 by @MatteoPierro performed one extra GraphQL API call. I improved upon this behaviour in f8b53f2. I think we are now good to merge since people can enable this feature on their own Vercel instance by adding the `FETCH_MULTI_PAGE_STARS=true` env variable while on the public instance for now, it is disabled. This env variable would also allow us to create a GitHub action-based version that is more accurate. 🚀 

@anuraghazra and @MatteoPierro can you please review the changes I made?

File: src/fetchers/stats-fetcher.js [hunk 1]
```diff
@@ -63,39 +85,43 @@ const fetcher = (variables, token) => {
...
+    } else {
+      stats.data.data.user.repositories.nodes.push(...repoNodes);
+    }
+
+    // Disable multi page fetching on public Vercel instance due to rate limits.
+    const repoNodesWithStars = repoNodes.filter(
+      (node) => node.stargazers.totalCount !== 0,
+    );
+    hasNextPage =
+      process.env.FETCH_MULTI_PAGE_STARS === "true"
```
[the remark this answers is not in the document]
Comment (rickstaa) [reply src/fetchers/stats-fetcher.js]: A great suggestion I applied your changes in [fc5fa79](https://github.com/anuraghazra/github-readme-stats/pull/2159/commits/fc5fa795c2cbdf588d2f71b09c295f94842ed7cd).

**Review by anuraghazra — APPROVED** [review]

**Pull request merged**